src/lib/chips/chip-list.ts

```diff
@@ -317,4 +508,150 @@ export class MdChipList implements AfterContentInit, OnDestroy {
 
     return false;
   }
+
+  _setSelectionByValue(value: any, isUserInput: boolean = true) {
+    this._clearSelection();
+    this.chips.forEach(chip => chip.deselect());
+    if (Array.isArray(value)) {
+      value.forEach(currentValue => this._selectValue(currentValue, isUserInput));
+    }
+    this._sortValues();
+  }
+
+  /**
+   * Finds and selects the chip based on its value.
+   * @returns Chip that has the corresponding value.
+   */
+  private _selectValue(value: any, isUserInput: boolean = true): MdChip | undefined {
+
+    const correspondingChip = this.chips.find(chip => {
+      return chip.value != null && this._compareWith(chip.value,  value);
+    });
+
+    if (correspondingChip) {
+      isUserInput ? correspondingChip.selectViaInteraction() : correspondingChip.select();
+      this._selectionModel.select(correspondingChip);
+    }
+
+    return correspondingChip;
+  }
+
+  private _initializeSelection(): void {
+    // Defer setting the value in order to avoid the "Expression
+    // has changed after it was checked" errors from Angular.
+    Promise.resolve().then(() => {
+      this._setSelectionByValue(this.ngControl ? this.ngControl.value : this._value, false);
+      this.stateChanges.next();
+    });
+  }
+
+  /**
+   * Deselects every chip in the list.
+   * @param skip Chip that should not be deselected.
+   */
+  private _clearSelection(skip?: MdChip): void {
+    this._selectionModel.clear();
+    this.chips.forEach(chip => {
+      if (chip !== skip) {
+        chip.deselect();
+      }
+    });
+    this.stateChanges.next();
+  }
+
+  /**
+   * Sorts the model values, ensuring that they keep the same
+   * order that they have in the panel.
+   */
+  private _sortValues(): void {
+    this._selectionModel.clear();
+
+    this.chips.forEach(chip => {
+      if (chip.selected) {
+        this._selectionModel.select(chip);
+      }
+    });
+    this.stateChanges.next();
+  }
+
+  /** Emits change event to set the model value. */
+  private _propagateChanges(): void {
+    let valueToEmit = this.selected.length === 0 ? null : this.selected.map(chip => chip.value);
+    this._value = valueToEmit;
+    this.change.emit(new MdChipListChange(this, valueToEmit));
+    this.valueChange.emit(valueToEmit);
+    this._onChange(valueToEmit);
+    this._changeDetectorRef.markForCheck();
+  }
+
+  _blur() {
+    if (!this.disabled) {
+      if (this._chipInput) {
+        // Check if focus moved to chip input. If not, do real on blur
+        setTimeout(() => {
+          if (!this.focused) {
+            this._onBlur();
+          }
+        });
+      } else {
+        this._onBlur();
+      }
+    }
+  }
+
+  _onBlur() {
```

Comment:
It's weird that there's a `_blur()` and an `_onBlur()`- could the names be more specific? Function descriptions would also help